mipengine/algorithm_flow_DTOs.py

```diff
@@ -0,0 +1,18 @@
+from typing import List, Union, Any
```

Comment:
What the caller of the post_algorithm(...) gets when the algorithm execution finishes is anyway a string representation of the TabularDataResult, in json format, so no objects are actually returned to the caller(the backend).
From my perspective the module `algorithm_flow_DTOs.py`(or `algorithm_result_DTOs.py` as it used to be named) contains dtos that we want to have available in the algorithm flow modules. Up until now the only class that the algorithm flow needed to have access to, was the TabularDataResult class, now adding the class Literal which has nothing to do with "results" made me change the name of the module. As for the reasons I added the Literal class I ll explain in the relevant comment..

Reply:
I understand the reason it was unified with `Literal` , it makes sense.

My point is that only the `Result` objects are actually DTOs that need to be jsonifiable for example (inherit from BaseModel). The `Literal` class for example has no actual need to be jsonifiable.

So in my mind these classes have differences in their usage and it would be easier to keep them separated.

Up to you.

Reply:
Inheriting from pydantic's BaseModel is not only used to give easy json export to an class, it is a more general and handy library for data structures management.